wds-cmb2-date-range-field.php

```diff
@@ -192,6 +193,51 @@ function sanitize( $override_value, $value ) {
 		return $value;
 
 	}
+	
+	/**
+	 * Save the start date and end date into separate fields if the argugment split_date_range is true
+	 *
+	 * @param string            $field_id The current field id paramater.
+	 * @param bool              $updated  Whether the metadata update action occurred.
+	 * @param string            $action   Action performed. Could be "repeatable", "updated", or "removed".
+	 * @param CMB2_Field object $field    This field object
+	 */
+	function save_split_date_range( $field_id, $updated, $action, $cmb2_field ) {
+
+		if ( ! $updated || $action == 'repeatable' ) {
+			return;
+		}
+
+		$split_values = $cmb2_field->args( 'split_date_range' );
+
+		if ( $split_values === true ) {
+			
+			$object_type  = $cmb2_field->object_type;
+			$object_id    = $cmb2_field->object_id;
+			$value        = $cmb2_field->value;
+
+			$start_date_key = $field_id . '_start';
+			$end_date_key   = $field_id . '_end';
+	
+			if ( $action == 'removed' ) {
+	
+				delete_metadata( $object_type, $object_id, $start_date_key );
+				delete_metadata( $object_type, $object_id, $end_date_key );
+	
+			} else {
+				$value = json_decode( $value, true );
+	
+				if ( is_array( $value ) ) {
+					$value = array_map( 'sanitize_text_field', $value );
+	
+					update_metadata( $object_type, $object_id, $start_date_key, $value['start'] );
+					update_metadata( $object_type, $object_id, $end_date_key, $value['end'] );
+					
```

Comment:
using these metadata functions directly does not account for all types of uses (e.g. `options-page` boxes). You should use the `CMB2_Field` methods `remove_data`/`update_data`.

Reply:
Updated on last commit where is used the `CMB2_Field` methods `remove_data` and `update_data`.

Reply:
This have now been updated